js/map.js

```diff
@@ -3,6 +3,36 @@ function htmlentities(str) {
   return String(str).replace(/&/g, '&amp;').replace(/</g, '&lt;').replace(/>/g, '&gt;').replace(/"/g, '&quot;');
 }
 $(document).ready(function() {
+  // main map cache
+  var cache = {};
+
+    window.addEventListener("message", function (evt) {
+      var data = JSON.parse(evt.data);
+      switch(data.cmd){
+        case 'catch_alert':
+          overpass.handlers["onAjaxError"] = function(errmsg) {
```

Comment:
We should probably store any previously set event handlers and call them inside the function?